### ISSUE TITLE
Understanding heuristic-based outlier detection

### ISSUE THREAD

**Issue by clevilll** [yzhao062/pyod#611]
I'm struggling to understand mathematics and relate to new cross-density-distance outlier detection algorithms that were newly published in the *Computer & Security* journal. I already created an [issue][1] in PyOD Repo and mentioned the general approach in this recent paper for further consideration if this approach is valid to compare with other algorithms in [tag:PyOD] pythonic package.  

**Background:**

I'm familiar with known [unsupervised anomaly/outlier detection][2]:
- density-based
- distance-based
- autoencoder
- model-based
- cross-density-distance

based on this [post][3] linked with [this][4]:
> *There is no such algorithm "which works in most cases". The task heavily depends on the specifics of your case, e.g. whether you need local anomalies when a point differs from other points near it or global ones when a point does not look similar to any other point in the dataset. A very good review of anomaly detection algorithms can be found in [here][5]*

> *"A Comparative Evaluation of Unsupervised Anomaly Detection Algorithms for Multivariate Data"* by Markus Goldstein, Seiichi Uchida

- I'm relatively familiar with general processes within unsupervised OD algorithms which end up with scoring outlierness so-called "**outlier score**" considering anomalous rate or contamination rate of 5% to 10% which set it up using the **threshold** for a related (hyper-)parameter i.e `contamination=0.05` 

    Based on my findings, I really liked the procedure suggested in the following book including 3 steps of modeling procedure for anomaly detection involves:

   1. model development
   2. threshold determination
   3. feature evaluation

    > "Handbook of Anomaly Detection: With Python Outlier Detection examples" by Chris Kuo, published in [Dataman in AI][6]
       
    Later they generate mock data with a certain contamination rate and *split data* into train-set and test-set

 - I'm also informed about **geometry** approaches being used in algorithms like *dimension reduction* algorithms like LDA and PCA using **projection** to find the best fit to separate observation of different classes

**My question\Concern:**

I saw for example how [ECOD][7] or [HBOS][8] works but the approach used in the discussed journal paper and its Math is vague and unclear and since I explained [issue][1] I avoid explaining but I will list my Questions:
 

- Q1. Does it make sense for one to calculate the outlier score and apply the confidence score and on top of it, we start allocating weight? This logic if we accept it with closed eyes, how could be indicated in the cubic form based on equation (#18) in the paper:
> *Compute the final ensemble score $O_i$ for data point $X_i$ passed through all submodels*
>  $$
 E(O, W)=\sum_{i=1}^n\left(s_i \cdot s_{c i} \cdot \omega_i\right)^3
 $$

- The author states after the equation (#15) in the paper (page #7):

>  Throughout our experiment, we set the **density score weight** ( $𝜔_{𝑑𝑒𝑛𝑠𝑖𝑡𝑦}$ ) to **0.8** and the **distance score weight** ($𝜔_{𝑑𝑖𝑠𝑡𝑎𝑛𝑐𝑒}$) to **0.2**.
It's not clear how the author comes up even empirically with these coefficients to weigh and mix scoring the way the user prefers or gets better accuracy if he has labels!

The impact of this approach is indicated in *Fig 4* in paper between density-based and distance-based scoring using **empirical** coefficients denoted in equation (#15) as weights or the impact of human assistance! 

- Q2. generally speaking, modifying some scoring using human factor, does not mean that manipulating scoring to get the best results while the art is finding non-parametric detection to minimize human factor parameterizations? 

Maybe I'm not fully informed and as an enthusiast want to understand the reliability and validity of this promising detection however if we are going towards *unsupervised* OD\AD then we should go towards **non-parametric** algorithms while setting proposed parameters ($𝜔_{𝑑𝑖𝑠𝑡𝑎𝑛𝑐𝑒}$) and ( $𝜔_{𝑑𝑒𝑛𝑠𝑖𝑡𝑦}$ ) as human-factored is not against of spirit of unsupervised algorithms and selling the ideas as Human-assisted model? 

 - Q3. which logic could explain (XAI or explainability) of detection scoring should be ( $𝜔_{𝑑𝑒𝑛𝑠𝑖𝑡𝑦}$ ) > ($𝜔_{𝑑𝑖𝑠𝑡𝑎𝑛𝑐𝑒}$) means intentionally we force ( $𝜔_{𝑑𝑒𝑛𝑠𝑖𝑡𝑦}$ ) will be dominant than ($𝜔_{𝑑𝑖𝑠𝑡𝑎𝑛𝑐𝑒}$)? Can one explain based on empirical experiments of **contextualized features** to use this trick specifically over imbalanced data which is no guaranteed if they are normal distribution in the form of Gaussian distribution?
 
This approach is sound's like ad-hoc'ery in outlier detection, as practiced *normally* by non-statisticians... [ref](https://stats.stackexchange.com/a/199844/240550)

[How to test unsupervised learning methods for anomaly detection?](https://datascience.stackexchange.com/q/97099/69263)

--------------

**Side notes:**
- Please see this [post][9] to understand the *"difference between Outlier and Anomaly in the context of machine learning"* 
- Please see these posts to understand OD\AD in the domain of **IT and network\cyber security** context: [post1][10] [post2][11] [post3][12] [post4][13]
- Please see this [post][14] to understand *"...anomaly detection should **not** be considered as classification problem"* or binary-classification
- Please see these posts to understand in OD\AD, **overfitting** is **not** solution: [post1][15] [post2][16]
- Please see this post to understand the **data split** for OD\AD models: [post1][17]
- see to understand how to select the **threshold** for unsupervised anomaly detection [post1][18] 
- understand OD\AD over imbalanced data or  heavy-tailed data or skewed data\distribution [post1][19] [post2][20]
  

  [1]: https://github.com/yzhao062/pyod/issues/610
  [2]: https://stats.stackexchange.com/a/474663/240550
  [3]: https://stackoverflow.com/a/38545674/10452700
  [4]: https://stats.stackexchange.com/q/625294/240550
  [5]: http://www.goldiges.de/publications/Comparison_Unsupervised_Anomaly_Detection.pdf
  [6]: https://medium.com/dataman-in-ai?source=post_page---byline--5cbf3e3021eb------------------
  [7]: https://medium.com/dataman-in-ai/handbook-of-anomaly-detection-with-python-outlier-detection-3-ecod-5cbf3e3021eb
  [8]: https://medium.com/dataman-in-ai/handbook-of-anomaly-detection-2-hbos-3f2d304977f8
  [9]: https://stats.stackexchange.com/q/189664/240550
  [10]: https://stats.stackexchange.com/a/23489/240550
  [11]: https://stats.stackexchange.com/q/141017/240550
  [12]: https://stats.stackexchange.com/a/450473/240550
  [13]: https://stats.stackexchange.com/q/268274/240550
  [14]: https://stats.stackexchange.com/a/592677/240550
  [15]: https://stats.stackexchange.com/q/330904/240550
  [16]: https://stats.stackexchange.com/q/598133/240550
  [17]: https://stats.stackexchange.com/q/572346/240550
  [18]: https://datascience.stackexchange.com/q/110205/69263
  [19]: https://stats.stackexchange.com/q/33620/240550
  [20]: https://stats.stackexchange.com/q/346774/240550
